python/python/glide/redis_client.py

```diff
@@ -194,6 +197,46 @@ async def _write_buffered_requests_to_socket(self) -> None:
         self._writer.write(b_arr)
         await self._writer.drain()
 
+    # TODO: change `str` to `TEncodable` where `TEncodable = Union[str, bytes]`
+    def _encode_arg(self, arg: str) -> bytes:
+        """
+        Converts a string argument to bytes.
+
+        Args:
+            arg (str): An encodable argument.
+
+        Returns:
+            bytes: The encoded argument as bytes.
+        """
+
+        # TODO: Allow passing different encoding options
+        return bytes(arg, encoding="utf8")
```

Comment:
It is not completely platform agnostic - we need to add handling in utf8 usage.
For example AIX and Solaris are unix based os that don't use utf8 in their default. It is possible to configure new versions to use utf8, but the old versions don't even have the option.
If at some point we'll want to support windows, it can be very problematic.
It might not be relevant enough at that point, but we need more than allowing the passing of different encoding. To avoid crashing or producing incorrect results, we need to check the OS we are running on when no specific encoding is passed.

Reply:
ATM we will support only utf8 strings and bytes as passed arguments and return values. Later on we will support encoder as a client configuration which will allow more encoding options. If the user doesn't use utf8 strings, he can use bytes instead